src/HyperfundFactory.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.28;
+
+import "./Hyperfund.sol"; // Import Hyperfund contract
+import "./Hyperstaker.sol"; // Import Hyperstaker contract
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+contract HyperfundFactory is Initializable, UUPSUpgradeable, OwnableUpgradeable {
+    // Mapping to associate (hypercert ID) with Hyperfund and Hyperstaker addresses
+    mapping(uint256 => address) public hyperfunds;
+    mapping(uint256 => address) public hyperstakers;
+
+    error InvalidHypercertMinter();
```

Comment:
can be just InvalidAddress :)

Reply:
We should specify which address is invalid

Reply:
why? if one is zero address it's ok i think